homeassistant/components/cover/__init__.py

```diff
@@ -363,6 +371,26 @@ async def async_toggle_tilt(self, **kwargs):
         else:
             await self.async_close_cover_tilt(**kwargs)
 
+    def async_write_ha_state(self) -> None:
+        """Set last toggle direction when writing ha state."""
+        super().async_write_ha_state()
+        if self.state == STATE_OPENING:
+            self._attr_is_last_toggle_direction_open = True
+        elif self.state == STATE_CLOSING:
+            self._attr_is_last_toggle_direction_open = False
+        # do nothing on all other states
```

Comment:
Instead of overriding `Entity.async_write_ha_state`, should the update of `self._attr_is_last_toggle_direction_open` be done in the `state` property instead?

Reply:
Done!